imageregistry/v1/types.go

```diff
@@ -0,0 +1,346 @@
+	// be associated with.
+	// +optional
+	ProjectID string `json:"projectID,omitempty" protobuf:"bytes,3,opt,name=projectID"`
+	// KeyID is the KMS key ID to use for encryption.
+	// Optional, buckets are encrypted by default on GCP.
+	// This allows for the use of a custom encryption key.
+	// +optional
+	KeyID string `json:"keyID,omitempty" protobuf:"bytes,4,opt,name=keyID"`
+}
+
+// ImageRegistryConfigStorageSwift holds the information to configure
+// the registry to use the OpenStack Swift service for backend storage
+// https://docs.docker.com/registry/storage-drivers/swift/
+type ImageRegistryConfigStorageSwift struct {
+	// +optional
+	AuthURL string `json:"authURL" protobuf:"bytes,1,opt,name=authURL"`
+	// +optional
+	AuthVersion string `json:"authVersion" protobuf:"bytes,2,opt,name=authVersion"`
+	// +optional
+	Container string `json:"container" protobuf:"bytes,3,opt,name=container"`
+	// +optional
+	Domain string `json:"domain" protobuf:"bytes,4,opt,name=domain"`
+	// +optional
+	DomainID string `json:"domainID" protobuf:"bytes,5,opt,name=domainID"`
+	// +optional
+	Tenant string `json:"tenant" protobuf:"bytes,6,opt,name=tenant"`
+	// +optional
+	TenantID string `json:"tenantID" protobuf:"bytes,7,opt,name=tenantID"`
+	// +optional
+	RegionName string `json:"regionName" protobuf:"bytes,8,opt,name=regionName"`
+}
+
+type ImageRegistryConfigStoragePVC struct {
+	// +optional
+	Claim string `json:"claim" protobuf:"bytes,1,opt,name=claim"`
+}
+
+// ImageRegistryConfigStorageAzure holds the information to configure
+// the registry to use Azure Blob Storage for backend storage.
+type ImageRegistryConfigStorageAzure struct {
+	// +optional
+	AccountName string `json:"accountName" protobuf:"bytes,1,opt,name=accountName"`
+	// +optional
+	Container string `json:"container" protobuf:"bytes,2,opt,name=container"`
+}
+
+// ImageRegistryConfigStorage describes how the storage should be configured
+// for the image registry.
+type ImageRegistryConfigStorage struct {
+	// EmptyDir represents ephemeral storage on the pod's host node.
+	// This storage cannot be used with more than 1 replica and is not suitable
+	// for production use. When the pod is removed from a node for any reason,
+	// the data in the emptyDir is deleted forever.
+	// This configuration is EXPERIMENTAL and is subject to change without notice.
+	// +optional
+	EmptyDir *ImageRegistryConfigStorageEmptyDir `json:"emptyDir,omitempty" protobuf:"bytes,1,opt,name=emptyDir"`
+	// S3 represents configuration that uses Amazon Simple Storage Service.
+	// +optional
+	S3 *ImageRegistryConfigStorageS3 `json:"s3,omitempty" protobuf:"bytes,2,opt,name=s3"`
+	// GCS represents configuration that uses Google Cloud Storage.
+	// +optional
+	GCS *ImageRegistryConfigStorageGCS `json:"gcs,omitempty" protobuf:"bytes,3,opt,name=gcs"`
+	// Swift represents configuration that uses OpenStack Object Storage.
+	// This configuration is EXPERIMENTAL and is subject to change without notice.
```

Comment:
@adambkaplan can we remove this EXPERIMENTAL notice?

Reply:
@ricardomaraschini @dmage definitely drop EXPERIMENTAL. Ditto `emptyDir` - it's GA as it's api stable.

Reply:
Done.